test_driver/fieldvalue_behaviors.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:async';
+import 'dart:convert' show utf8;
+import 'dart:io';
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:cloud_firestore_mocks/cloud_firestore_mocks.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:test/test.dart' as _test;
+
+import 'firestore_clients.dart';
+
+/// Test cases to compare FieldValue implementation behaviors. This test
+/// requires 2 invocations. One for cloud_firestore_mocks and the other for
+/// real Firestore and Firestore Emulator backend.
+///
+/// Background: cloud_firestore_mocks overwrites
+/// FieldValueFactoryPlatform.instance to use a customized FieldValueFactory.
+/// The instance field updates `static final` _factory of FieldValue class.
+/// Because the field cannot be updated within one Dart runtime, we need have
+/// two invocations for the same set of assertion against cloud_firestore
+/// and cloud_firestore_mocks.
+void main() {
+  String firestoreImplementation =
+      Platform.environment['FIRESTORE_IMPLEMENTATION'];
+  print('firestoreImplementation on device $firestoreImplementation');
```

Comment:
This `firestoreImplementation` is always null.

Reply:
Resolved. Now I use requestData between driver program and test application on a device.